packages/apps/fortune/exchange-oracle/server/src/modules/job/job.service.ts

```diff
@@ -131,23 +138,22 @@ export class JobService {
     if (!recordingOracleWebhookUrl)
       throw new NotFoundException('Unable to get Recording Oracle webhook URL');
 
-    if (
-      this.storage[escrowAddress] &&
-      this.storage[escrowAddress].includes(workerAddress)
-    )
-      throw new BadRequestException('User has already submitted a solution');
-
-    if (!this.storage[escrowAddress]) {
-      this.storage[escrowAddress] = [];
-    }
-    this.storage[escrowAddress].push(workerAddress);
+    const solutionUrl = await uploadJobSolutions(
+      this.minioClient,
+      chainId,
+      escrowAddress,
+      workerAddress,
+      signer.address,
+      solution,
+      this.s3Config.bucket,
+    );
 
     await this.httpService.post(recordingOracleWebhookUrl, {
       escrowAddress: escrowAddress,
       chainId: chainId,
       exchangeAddress: signer.address,
```

Comment:
I think you don't need exchangeAddress here, it is already in the file and when we add the security it should be in the signature